src/opera/commands/validate.py

```diff
@@ -40,43 +40,46 @@ def _parser_callback(args):
         print("Invalid inputs: {}".format(e))
         return 1
 
+    if args.csar_or_service_template is None:
+        csar_or_st_path = PurePath(".")
+    else:
+        csar_or_st_path = PurePath(args.csar_or_service_template)
+
     try:
-        if is_zipfile(args.csar.name):
+        if is_zipfile(csar_or_st_path) or Path(csar_or_st_path).is_dir():
             print("Validating CSAR...")
-            validate_compressed_csar(args.csar.name, inputs)
+            validate_csar(csar_or_st_path, inputs)
         else:
             print("Validating service template...")
-            validate_service_template(args.csar.name, inputs)
+            validate_service_template(csar_or_st_path, inputs)
         print("Done.")
     except ParseError as e:
         print("{}: {}".format(e.loc, e))
         return 1
-    except DataError as e:
+    except OperaError as e:
         print(str(e))
         return 1
 
     return 0
 
 
-def validate_compressed_csar(csar_name: str, inputs: typing.Optional[dict]):
+def validate_csar(csar_path: PurePath, inputs: typing.Optional[dict]):
     if inputs is None:
         inputs = {}
 
-    with TemporaryDirectory() as csar_validation_dir:
-        csar = CloudServiceArchive.create(PurePath(csar_name))
-        csar.validate_csar()
-        tosca_service_template = csar.get_entrypoint()
-
-        # unzip csar to temporary folder
-        ZipFile(csar_name, "r").extractall(csar_validation_dir)
+    csar = CloudServiceArchive.create(csar_path)
+    csar.validate_csar()
+    entrypoint = csar.get_entrypoint()
 
-        # try to initiate service template from csar
-        ast = tosca.load(Path(csar_validation_dir), Path(tosca_service_template))
-        ast.get_template(inputs)
+    if entrypoint is not None:
+        with TemporaryDirectory() as csar_validation_dir:
+            # unzip csar to temporary folder
+            ZipFile(csar_path, "r").extractall(csar_validation_dir)
```

Comment:
If we're validating a compressed TOSCA CSAR this won't go through because it would try to extract a directory.  I tried to run this locally and got:

```console
(.venv) anzoman@ubuntu:~/Desktop/xopera-opera/testing$ ls -l
total 12
drwxrwxr-x 10 anzoman anzoman 4096 May 12 08:53 modules
-rw-rw-r--  1 anzoman anzoman 3840 Apr  9 09:36 service-template.yaml
drwxrwxr-x  2 anzoman anzoman 4096 May 12 08:53 TOSCA-Metadata
(.venv) anzoman@ubuntu:~/Desktop/xopera-opera/testing$ opera validate
Validating CSAR...
Traceback (most recent call last):
  File "/home/anzoman/Desktop/xopera-opera/.venv/bin/opera", line 33, in <module>
    sys.exit(load_entry_point('opera', 'console_scripts', 'opera')())
  File "/home/anzoman/Desktop/xopera-opera/src/opera/cli.py", line 47, in main
    return args.func(args)
  File "/home/anzoman/Desktop/xopera-opera/src/opera/commands/validate.py", line 51, in _parser_callback
    validate_csar(csar_or_st_path, inputs)
  File "/home/anzoman/Desktop/xopera-opera/src/opera/commands/validate.py", line 77, in validate_csar
    ZipFile(csar_path, "r").extractall(csar_validation_dir)
  File "/usr/local/lib/python3.6/zipfile.py", line 1113, in __init__
    self.fp = io.open(file, filemode)
IsADirectoryError: [Errno 21] Is a directory: '.'
```

You should probably just check if the `csar_path` is compressed and then do the extracting part in that case or if it's a dir you should copy its files to the `csar_validation_dir`.

Reply:
Fixed!

Reply:
Hmm weird - I'm still having some problems:

```console
(.venv) anzoman@ubuntu:~/Desktop/xopera-opera/testing$ ls -l
total 12
drwxrwxr-x 10 anzoman anzoman 4096 May 13 08:28 modules
-rw-rw-r--  1 anzoman anzoman 3840 Apr  9 09:36 service-template.yaml
drwxrwxr-x  2 anzoman anzoman 4096 May 13 08:28 TOSCA-Metadata
(.venv) anzoman@ubuntu:~/Desktop/xopera-opera/testing$ opera validate
Validating CSAR...
Traceback (most recent call last):
  File "/home/anzoman/Desktop/xopera-opera/.venv/bin/opera", line 33, in <module>
    sys.exit(load_entry_point('opera', 'console_scripts', 'opera')())
  File "/home/anzoman/Desktop/xopera-opera/src/opera/cli.py", line 47, in main
    return args.func(args)
  File "/home/anzoman/Desktop/xopera-opera/src/opera/commands/validate.py", line 51, in _parser_callback
    validate_csar(csar_or_st_path, inputs)
  File "/home/anzoman/Desktop/xopera-opera/src/opera/commands/validate.py", line 77, in validate_csar
    ZipFile(csar_path, "r").extractall(csar_validation_dir)
  File "/usr/local/lib/python3.6/zipfile.py", line 1113, in __init__
    self.fp = io.open(file, filemode)
IsADirectoryError: [Errno 21] Is a directory: '.'
```

Reply:
Sorry my bad - I didn't pull your latest changes. It works okay now.